src/libostree/ostree-repo-commit.c

```diff
@@ -168,6 +172,74 @@ ot_security_smack_reset_fd (int fd)
 #endif
 }
 
+gboolean
+_ostree_tmpf_fsverity (OstreeRepo *self,
+                       GLnxTmpfile *tmpf,
+                       GError    **error)
+{
+#ifdef HAVE_LINUX_FSVERITY_H
+  _OstreeFeatureSupport fsverity_enabled;
+  g_mutex_lock (&self->txn_lock);
+  fsverity_enabled = self->fs_verity;
+  g_mutex_unlock (&self->txn_lock);
+
+  if (fsverity_enabled == _OSTREE_FEATURE_NO)
+    {
+      if (G_UNLIKELY(self->fsverity_required))
+        return glnx_throw (error, "fsverity required but filesystem does not support it");
+      return TRUE;
+    }
+
+  GLNX_AUTO_PREFIX_ERROR ("fsverity", error);
+
+  /* fs-verity requires a read-only file descriptor */
+  if (!glnx_tmpfile_reopen_rdonly (tmpf, error))
+    return FALSE;
+
+  struct fsverity_enable_arg arg = { 0, };
+  
+  arg.version = 1;
+  arg.hash_algorithm =  FS_VERITY_HASH_ALG_SHA256;  /* TODO configurable */
+  arg.block_size = 4096; /* FIXME query */
+  arg.salt_size = 0; /* TODO store salt in ostree repo config */
+  arg.salt_ptr = 0;
+  arg.sig_size = 0; /* TODO use signatures?  Or maybe just tell people to do this after */
+  arg.sig_ptr = 0;
+
+  if (ioctl (tmpf->fd, FS_IOC_ENABLE_VERITY, &arg) < 0)
+    {
+      switch (errno)
+        {
+          case ENOTTY:
+          case EOPNOTSUPP:
+            if (!self->fsverity_required)
+              {
+                /* If we got here, we must be trying "opportunistic" use of fs-verity */
+                g_assert_cmpint (fsverity_enabled, ==, _OSTREE_FEATURE_MAYBE);
+                g_mutex_lock (&self->txn_lock);
+                self->fs_verity = _OSTREE_FEATURE_NO;
+                g_mutex_unlock (&self->txn_lock);
+                return TRUE;
+              }
+            else
+              {
+                return glnx_throw_errno_prefix (error, "(config required) ioctl(FS_IOC_ENABLE_VERITY)");
```

Comment:
Or let's just throw the same error as we would've above? `fsverity required but filesystem does not support it`

Reply:
Minor: could reorg this to return upfront and de-indent